src/bucket/BucketManagerImpl.cpp

```diff
@@ -486,11 +516,39 @@ BucketManagerImpl::renameBucketDirFile(std::filesystem::path const& src,
     }
 }
 
-std::shared_ptr<Bucket>
+template <>
+std::shared_ptr<LiveBucket>
+BucketManager::adoptFileAsBucket(BucketManager& bm, std::string const& filename,
+                                 uint256 const& hash, MergeKey* mergeKey,
+                                 std::unique_ptr<BucketIndex const> index)
+{
+    auto& bmImpl = static_cast<BucketManagerImpl&>(bm);
```

Comment:
i think we wouldn't need to do this if we didn't follow the impl-class pattern right?